### ISSUE TITLE
Error in find_gene_modules function

### ISSUE THREAD

**Issue by sanjayrsrivatsan** [cole-trapnell-lab/monocle3#357]
**Describe the bug**

When ordering genes are used in preprocess_cds, find_gene_modules throws the following error:
`Error in preprocess_mat[rownames(cds), ] : subscript out of bounds`

**sessionInfo():**
Run sessionInfo() in R and post the output
```R version 3.6.1 (2019-07-05)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Mojave 10.14.5

Matrix products: default
BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] shiny_1.4.0                 data.table_1.12.6          
 [3] monocle3_0.2.0              SingleCellExperiment_1.8.0 
 [5] SummarizedExperiment_1.16.0 DelayedArray_0.12.0        
 [7] BiocParallel_1.20.0         matrixStats_0.55.0         
 [9] GenomicRanges_1.38.0        GenomeInfoDb_1.22.0        
[11] IRanges_2.20.0              S4Vectors_0.24.0           
[13] Biobase_2.46.0              BiocGenerics_0.32.0        
[15] RColorBrewer_1.1-2          devtools_2.2.1             
[17] usethis_1.5.1               ggridges_0.5.1             
[19] viridis_0.5.1               viridisLite_0.3.0          
[21] forcats_0.4.0               stringr_1.4.0              
[23] dplyr_0.8.3                 purrr_0.3.3                
[25] readr_1.3.1                 tidyr_1.0.0                
[27] tibble_2.1.3                ggplot2_3.2.1              
[29] tidyverse_1.2.1            

loaded via a namespace (and not attached):
  [1] readxl_1.3.1             backports_1.1.5         
  [3] plyr_1.8.4               igraph_1.2.4.1          
  [5] lazyeval_0.2.2           splines_3.6.1           
  [7] sp_1.3-2                 crosstalk_1.0.0         
  [9] scater_1.14.3            digest_0.6.22           
 [11] htmltools_0.4.0          rsconnect_0.8.15        
 [13] gdata_2.18.0             fansi_0.4.0             
 [15] magrittr_1.5             memoise_1.1.0           
 [17] limma_3.42.0             remotes_2.1.0           
 [19] modelr_0.1.5             gmodels_2.18.1          
 [21] prettyunits_1.0.2        colorspace_1.4-1        
 [23] rvest_0.3.5              ggrepel_0.8.1           
 [25] haven_2.2.0              callr_3.3.2             
 [27] crayon_1.3.4             RCurl_1.95-4.12         
 [29] jsonlite_1.6             zeallot_0.1.0           
 [31] glue_1.3.1               gtable_0.3.0            
 [33] zlibbioc_1.32.0          XVector_0.26.0          
 [35] pkgbuild_1.0.6           BiocSingular_1.2.0      
 [37] scales_1.1.0             DBI_1.0.0               
 [39] Rcpp_1.0.3               xtable_1.8-4            
 [41] spData_0.3.2             units_0.6-5             
 [43] spdep_1.1-3              rsvd_1.0.2              
 [45] proxy_0.4-23             htmlwidgets_1.5.1       
 [47] httr_1.4.1               ellipsis_0.3.0          
 [49] pkgconfig_2.0.3          farver_2.0.1            
 [51] deldir_0.1-23            uwot_0.1.8              
 [53] utf8_1.1.4               tidyselect_0.2.5        
 [55] labeling_0.3             rlang_0.4.1             
 [57] reshape2_1.4.3           later_1.0.0             
 [59] pbmcapply_1.5.0          munsell_0.5.0           
 [61] cellranger_1.1.0         tools_3.6.1             
 [63] cli_1.1.0                generics_0.0.2          
 [65] broom_0.5.2              batchelor_1.2.2         
 [67] fastmap_1.0.1            yaml_2.2.0              
 [69] processx_3.4.1           fs_1.3.1                
 [71] RANN_2.6.1               nlme_3.1-142            
 [73] mime_0.7                 xml2_1.2.2              
 [75] compiler_3.6.1           rstudioapi_0.10         
 [77] beeswarm_0.2.3           plotly_4.9.1            
 [79] e1071_1.7-2              testthat_2.3.0          
 [81] stringi_1.4.3            ps_1.3.0                
 [83] desc_1.2.0               RSpectra_0.15-0         
 [85] lattice_0.20-38          Matrix_1.2-17           
 [87] classInt_0.4-2           vctrs_0.2.0             
 [89] LearnBayes_2.15.1        pillar_1.4.2            
 [91] lifecycle_0.1.0          RcppAnnoy_0.0.13        
 [93] BiocNeighbors_1.4.1      bitops_1.0-6            
 [95] irlba_2.3.3              httpuv_1.5.2            
 [97] R6_2.4.0                 promises_1.1.0          
 [99] KernSmooth_2.23-16       gridExtra_2.3           
[101] vipor_0.4.5              sessioninfo_1.1.1       
[103] codetools_0.2-16         gtools_3.8.1            
[105] MASS_7.3-51.4            boot_1.3-23             
[107] assertthat_0.2.1         pkgload_1.0.2           
[109] leidenbase_0.1.0         rprojroot_1.3-2         
[111] withr_2.1.2              GenomeInfoDbData_1.2.2  
[113] expm_0.999-4             hms_0.5.2               
[115] grid_3.6.1               coda_0.19-3             
[117] class_7.3-15             DelayedMatrixStats_1.8.0
[119] sf_0.8-0                 lubridate_1.7.4         
[121] ggbeeswarm_0.6.0        
```

**Comment by mwdorrity** [comment]
Hi, a fix here to ensure your gene ids are all found in the gene loadings of your cds:

```
pr_deg_ids_fix = intersect(pr_deg_ids, rownames(cds@preprocess_aux$gene_loadings))

gene_module_df =
  find_gene_modules(mesoderm_cds[pr_deg_ids_fix,], 
  resolution=1e-2,umap.fast_sgd = F,
  random_seed = 42)
```

**Comment by brgew** [comment]
Hi Sanjay,
Thank you for the bug report.
I am unfamiliar with ordering genes. Could you please show me the commands that you used to get this error? I want to try to reproduce it.
Thank you.
Brent

**Comment by brgew** [comment]
Hi Sanjay,
I see that this issue was fixed in version 0.2.1, the current master branch, so I will close this issue.
